### PR TITLE
feature/95: Added integration tests

### LINKED ISSUE #95: Add integration tests
Currently, this project is only integration tested by the repos implementing it.
Solution:
Add an integration test that tests the functionality of this repo using a mock implementation.

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #95

### CODE REVIEW

File: parentPom.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.exasol</groupId>
+    <artifactId>virtual-schema-common-document-parent</artifactId>
+    <packaging>pom</packaging>
+    <version>${revision}</version>
+    <properties>
+        <revision>6.1.0-SNAPSHOT</revision>
```
Comment (kaklakariada) [inline parentPom.xml:10]: We need to remember to remove the `-SNAPSHOT` ;)

File: test-project/mock-project/pom.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>virtual-schema-common-document-mock-project</artifactId>
+    <version>not-versioned</version>
+    <name>mock project</name>
```
Comment (kaklakariada) [inline test-project/mock-project/pom.xml:7]: 👍

File: test-project/mock-project/src/main/resources/META-INF/services/com.exasol.adapter.AdapterFactory [hunk 3]
```diff
@@ -0,0 +1 @@
+com.exasol.adapter.document.moc.FixedDataAdapterFactory
```
Comment (kaklakariada) [inline test-project/mock-project/src/main/resources/META-INF/services/com.exasol.adapter.AdapterFactory:1]:
```suggestion
com.exasol.adapter.document.mock.FixedDataAdapterFactory
```

Do you mean `mock`? Then you will also need to rename the package in the java sources.

File: src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    @BeforeAll
+    static void beforeAll() throws SQLException, BucketAccessException, TimeoutException, FileNotFoundException {
+        testSetup = new ExasolTestcontainerTestSetup();
+        final UdfTestSetup udfTestSetup = new UdfTestSetup(testSetup);
+        exasolObjectFactory = new ExasolObjectFactory(testSetup.createConnection(),
+                ExasolObjectConfiguration.builder().withJvmOptions(udfTestSetup.getJvmOptions()).build());
+        buildMockAdapter();
+        final ExasolSchema adapterSchema = exasolObjectFactory.createSchema("ADAPTER");
+        testSetup.getDefaultBucket().uploadFile(Path.of("test-project", "mock-project", "target", MOCK_ADAPTER_JAR),
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java:45]:
```suggestion
        testSetup.getDefaultBucket().uploadFile(Path.of("test-project/mock-project/target", MOCK_ADAPTER_JAR),
```
This is more readable.

File: src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            mvnRunner.setSystemProperty("skipTests", "true");
+            mvnRunner.setSystemProperty("maven.test.skip", "true");
+            mvnRunner.setSystemProperty("ossindex.skip", "true");
+            mvnRunner.setSystemProperty("maven.javadoc.skip", "true");
+            mvnRunner.setSystemProperty("lombok.delombok.skip", "true");
+            mvnRunner.addCliOption("-PalternateTargetDir");
+            mvnRunner.executeGoal("package");
+            LOGGER.info("Done building mock-project");
+        } catch (final VerificationException exception) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-VSD-76")
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java:82]: Is it necessary to use an error ID in the integration test? The end user should never see this unless they run the tests themselves.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java (hunk 4 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java:45]: is it also platform independant?

File: src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java (hunk 5 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java:82]: Yes but we do it still. Is not only for end users but also for us. For example, it makes debugging easier if you know which error message you got.

File: parentPom.xml (hunk 1 above)
Comment (jakobbraun) [reply parentPom.xml:10]: yes

File: src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java (hunk 4 above)
Comment (kaklakariada) [reply src/test/java/com/exasol/adapter/document/DocumentAdapterIT.java:45]: Yes. I always used it also under windows.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**